test/e2e/daemonset/daemonset.go

```diff
@@ -0,0 +1,43 @@
+package daemonset
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DaemonSetCommands struct{ daemonSet appsv1.DaemonSet }
+
+func (dc DaemonSetCommands) ReadyConditionEventually() AsyncAssertion {
+	return Eventually(func() bool {
+		return dc.daemonSet.Status.DesiredNumberScheduled == dc.daemonSet.Status.NumberAvailable
+	}, 60*time.Second, 1*time.Second)
+}
+
+func DaemonSet(daemonSetKey types.NamespacedName) DaemonSetCommands {
```

Comment:
Let's rename it `Eventually` and return `AsyncAssertion` and since we have a custom matcher we just use the matcher with the `AsyncAssertion` so users will do:

```golang
daemonset.Eventually(DefaultNMStateHandler).Should(daemonset.BeReady())
```

This way the daemonset will be re-ready if `BeReady` fails and also we can negate it without extra code

```golang
daemonset.Eventually(DefaultNMStateHandler).ShouldNot(daemonset.BeReady())
```
This this in mind this functions should be:

```golang
func Eventually(daemonSetKey types.NamespacedName) AsyncAssertion {
return Eventually(func() appsv1.DaemonSet {
               daemonSet := appsv1.Daemonset{}
		_ = framework.Global.Client.Get(context.TODO(), daemonSetKey, &daemonSet)
		return daemonSet
}, 60*time.Second, 1*time.Second)
```